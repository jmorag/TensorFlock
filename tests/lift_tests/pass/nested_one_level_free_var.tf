main = f z 42;

z : Nat;
z = 3;

f : Nat -> Nat -> Nat;
f x j = j + k 3; {
  k : Nat -> Nat;
  k l = l + x;

  p : Nat;
  p = 15;
}
