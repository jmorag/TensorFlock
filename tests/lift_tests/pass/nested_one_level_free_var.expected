main = f : Nat -> Nat -> Nat[ z : Nat,42 : Nat] : Nat
(f x~2 j~1 : Nat -> Nat -> Nat) = j~1 : Nat + k : Nat -> Nat[ x~2 : Nat,3 : Nat] : Nat : Nat
{
(p  : Nat) = 15 : Nat
{

}
}
(z  : Nat) = 3 : Nat
{

}
(k x~2 l : Nat -> Nat -> Nat) = l : Nat + x~2 : Nat : Nat
{

}